### PR TITLE
Cloud computing cost optimization with cloud bursting

### LINKED ISSUE #775: Cost Optimization with Cloud Bursting
### Brief Summary: 
Describe the what, why, and how of your content idea in 2-5 sentences.

Cloud bursting is defined as a configuration set up between the public cloud and the private cloud to deal with the peaks in IT demands with the advantage of saving on costs. When resource requirements exceed what can be provided by the on-premise or collocated data center, the cloud bursting application will temporarily shift data into a public cloud to free up space in the primary data center.
### Key Takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?

What is Cloud Bursting 
Benefits of Cloud Bursting
Challenges of Cloud Bursting 
How to save money with cloud bursting
### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)
N/A

### PR DESCRIPTION
**Pull request by EKahuha**
this fixes #775 
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

File: articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+Third-party providers offer computing services over the public internet, known as the [public cloud](https://www.citrix.com/glossary/what-is-public-cloud.html). Public cloud computing services are either free or sold on-demand and thus readily available at low costs.
+
+On the other hand, the [private cloud](https://www.citrix.com/glossary/what-is-private-cloud.html) refers to computing services offered on a private internal network or over the internet to only selected users rather than the general public. Private clouds provide benefits such as self-service, elasticity, and scalability.
+
+Usually, you would need to overbuild physical infrastructure to increase your capacity to meet fluctuating demand peaks. Luckily, cloud bursting eliminates the need for all this by ensuring that private clouds are affordable. It does this by rightsizing private clouds in terms of storage and computes to accommodate rising demands and ensuring that the public cloud handles the peaks through a pay-as-you-use subscription.
```
Comment (lalith1403) [inline articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md:17]: The algorithm that decides the amount of compute power allocation is pretty interesting. You may want to touch upon it here.

File: articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+### Challenges of cloud bursting
+
+#### Managing configurations
+
+Organizations that cloud burst run the same resource types, for example, application servers, in multiple clouds. But choosing multiple cloud providers may bring some challenges in terms of different resource offerings. For instance, it may require an organization to manage and configure its applications and technology stacks across two environments.
+
+Differences may come in terms of cloud network configurations. Some cloud providers offer the concept of availability zones, while others do not. Still, some have security groups, others have Access Control Lists ([ACLs](https://geek-university.com/ccna/what-is-acl-access-control-list/#)) and [Subnets](https://en.wikipedia.org/wiki/Subnetwork#), and some have both.
+
+[Hypervisors](https://www.vmware.com/topics/glossary/content/hypervisor#), their features, and versions of hypervisors differ between cloud providers. Also, machine types and power vary from one cloud provider to another. Clouds also offer different storage types, including object, persistent, and block.
```
Comment (lalith1403) [inline articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md:43]: What are hypervisors? Attach links only when you feel additional information is provided at that link and is out of the scope of this article. To understand this article, the reader needs to know what Hypervisors are. As mentioned earlier, if the reader doesn't get information in the article, he loses his attention. Make articles reader-friendly by providing all the information needed. You may include a keywords section as mentioned in the previous review if required.

Comment (lalith1403) [reply articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md:43]: The same holds true with ACLs and subnets.

File: articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+Such differences may bring a downstream effect. For instance, for an organization to use different storage subsystems and hypervisors, it also has to use additional virtual machine images(https://www.informit.com/articles/article.aspx?p=1927741&seqNum=7) built for each cloud. Still, the organization has to maintain these base machine images with every security patch and operating system.
+
+#### Complexity and Latency
+
+Effective cloud bursting requires a flexible and scalable network that can handle the workload changes between public cloud infrastructures and on-premises private clouds. The management and [orchestration](https://www.mulesoft.com/resources/esb/what-application-orchestration#) of the applications moving between the two cloud environments must be in synchronization. And this requires two identical clouds with matching platforms and templates and a lot of scripting.
```
Comment (lalith1403) [inline articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md:49]: Explanation for orchestration. In networking, orchestration has various meanings on various layers of the OSI model.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Whenever a concept is included, you can either give a one-liner about it or explain it in the keywords section. Avoid using links to guide the reader. Do mention links, but don't expect the reader to learn the concept from the link. 

Well structured and well-written article. Once the changes are made, this will be ready for the final round of review.

File: articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md (hunk 1 above)
Comment (EKahuha) [reply articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md:17]: > compute power allocation
@lalith1403 
Which is this algorithm, please. I don't seem to understand this.

Comment (lalith1403) [reply articles/cloud-computing-cost-optimization-with-cloud-bursting/index.md:17]: You may ignore this for now. Btw, the Algorithms are called resource allocation algorithms.

**Comment by EKahuha** [comment]
@lalith1403 
Updated

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@EKahuha 
I'm noticing that this PR has merge conflicts, if you could please resubmit this (finished) PR on a new PR. 

I think the extra hero image was the error - I tried to delete it but that didn't work. 

image

Thank you

**Pull request closed without merging**